### ISSUE TITLE
MinGW (Windows): build error

### ISSUE THREAD

**Issue by opencv-pushbot** [Itseez/opencv#4778]
Transferred from http://code.opencv.org/issues/3785
```
|| Konstantin Kulakov on 2014-07-01 04:46
|| Priority: Low
|| Affected: branch 'master' (3.0-dev)
|| Category: build/install
|| Tracker: Bug
|| Difficulty: Medium
|| PR: 
|| Platform: x64 / Windows
```

Compilation (installation) error of opencv at WIndows
-----------
```
Sourse in github: https://github.com/itseez/opencv 
CMake: 3.0 
GCC: gcc (i686-posix-dwarf-rev2, Built by MinGW-W64 project) 4.9.0

I compiled OpenCV. I got an error:

[ 29%] Built target pch_Generate_opencv_core
Linking CXX shared library ..\..\bin\libopencv_core300.dll
C:/mingw/mingw32/bin/../lib/gcc/i686-w64-mingw32/4.9.0/../../../../i686-w64-ming
w32/bin/ld.exe: *cannot find -lRunTmChk*
collect2.exe: error: ld returned 1 exit status
modules\core\CMakeFiles\opencv_core.dir\build.make:1103: recipe for target 'bin/
libopencv_core300.dll' failed
mingw32-make[2]: *** [bin/libopencv_core300.dll] Error 1
CMakeFiles\Makefile2:1313: recipe for target 'modules/core/CMakeFiles/opencv_cor
e.dir/all' failed
mingw32-make[1]: *** [modules/core/CMakeFiles/opencv_core.dir/all] Error 2
makefile:135: recipe for target 'all' failed
mingw32-make: *** [all] Error 2
```

History
-------
##### Steven Puttemans on 2014-07-31 12:26
```
Did you have a previous VS system configured? I have read something about VC and Mingw screwing each others installations, and this seems exactly that.
```
##### Maxime Thébault on 2014-08-20 17:13
```
I'm having the same issue (cmake 2.8.12.2, Mingw 4.9.1 i686-posix-dwarf-rev0).

I think I've never had Visual Studio installed on that computer.
```
##### Florian Simond on 2014-08-24 17:19
```
I'm having the same problem.
I have never had VS installed on my computer neither, moreover I can compile OpenCV 2.4.9 without any problem...
```
##### lorenzoo lorenzoo on 2014-09-05 08:47
```
Hi all,

exactly the same here 

Windows7 64 bit and VS installed

By the way I can compile opencv 2.4.9 without problem
```
##### Ryein Goddard on 2014-09-11 21:41
```
I have the same issue using mingw as well.  I found that for some reason IPP was turn on by default when using Cmake 3.0 GUI.  I just disabled it and everything compiled fine.

I have AMD CPU & GPU so not sure why IPP was turned on.  IPP is Intel only right?
```
##### Florian Simond on 2014-09-12 06:11
```
Yes the problem seems to come with IPP.

Here is some line of OpenCVFindIPP.cmake:
<pre>
if(WIN32 AND MINGW AND NOT IPP_VERSION_MAJOR LESS 7)
    # Since IPP built with Microsoft compiler and /GS option
    # ======================================================
    # From Windows SDK 7.1
    #   (usually in "C:\Program Files\Microsoft Visual Studio 10.0\VC\lib"),
    # to avoid undefined reference to __security_cookie and _chkstk:
    set(MSV_RUNTMCHK "RunTmChk")
    set(IPP_LIBRARIES ${IPP_LIBRARIES} ${MSV_RUNTMCHK}${IPP_LIB_SUFFIX})

    # To avoid undefined reference to _alldiv and _chkstk
    # ===================================================
    # NB: it may require a recompilation of w32api (after having modified
    #     the file ntdll.def) to export the required functions
    #     See http://code.opencv.org/issues/1906 for additional details
    set(MSV_NTDLL    "ntdll")
    set(IPP_LIBRARIES ${IPP_LIBRARIES} ${MSV_NTDLL}${IPP_LIB_SUFFIX})
endif()
</pre>

I'm not a professional of CMake, but from what I understand it requires some resources of MSVC.... Am I correct ?
```
##### Michele Adduci on 2014-11-04 13:59
```
WITH_IPP=OFF, I can compile OpenCV 3.0 with MSVC 2012 without any problem, using OpenNI2 support and OpenGL too.

-   % Done changed from 0 to 100
```
##### Maksim Shabunin on 2015-04-28 14:34
```
-   Target version changed from 3.0 to 3.1
-   Priority changed from Normal to Low
```

**Comment by Goddard** [comment]
Still an issue on mingw 64bit on windows 8 with intel i7 4720HQ

**Comment by chinadragon0515** [comment]
I met the same issues with mingw, do we have a solution now? Do we have a target date of opencv3.1? thanks

**Comment by alalek** [comment]
@chinadragon0515 You can try to use `WITH_IPP=OFF` CMake parameter.

**Comment by mshabunin** [comment]
"Fixed" in #5282 by disabling IPP by default for mingw builds. It can still be enabled b providing `-DWITH_IPP=ON` option to cmake.

**Comment by AndreasWegmann** [comment]
That is not really a fix. You just disabled the IPP for mingw. When du you support this for mingw?
Is that really a problem of mingw or are there just missing references to be solved without visual studio ?

I like to use that for a performance critical application, so I'm really interested in a usable solution.
